File: backend/demo1/src/main/java/com/example/demo/entity/TimeSlot.java
package com.example.demo.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;
@Entity
@Table(name = "time_slots")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class TimeSlot {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;

    @Column(columnDefinition = "time",  nullable = false)
    @NotNull(message = "Start time can not be null")
    LocalTime startTime;

    @Column(columnDefinition = "time",  nullable = false)
    @NotNull(message = "End time can not be null")
    LocalTime endTime;

    @CreationTimestamp
    LocalDateTime createdAt;

    @UpdateTimestamp
    LocalDateTime updatedAt;

    @OneToMany(mappedBy = "timeSlot")
    private List<RoomReservation> reservations;
}

File: backend/backend-dotnet/Models/TimeSlot.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models;

[Table("time_slots")]
public class TimeSlot
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public TimeOnly StartTime { get; set; }

    [Required]
    public TimeOnly EndTime { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public ICollection<RoomReservation> Reservations { get; set; } = new List<RoomReservation>();
}
